src/MergeContext.js

```diff
@@ -954,12 +968,47 @@ class PullRequest {
         return false;
     }
 
+    async _getMergeCommit() {
+        const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+        return await GH.getCommit(mergeSha);
+    }
+
+    // recalculates this._stagingFailed value for PRs without staging commit
+    async _recalculateStagingFailed() {
+        assert(!this._stagedSha());
+
+        const allEvents = await GH.getIssueEvents(this._prNumber());
+        // we consider all commits created by the bot user and referencing this PR as staged commits
+        let stagedEvents = allEvents.filter(ev => ev.event === "referenced" && ev.actor.login === Config.githubUserLogin());
+        if (!stagedEvents.length)
+            return;
```

Comment:
Should not we recalculate in this case, as promised by the function name/description?

```suggestion
        if (!stagingEvents.length) {
            this._stagingFailed = false;
            return;
        }
```